nodes/test/test_rate_limiter.py

```
@@ -0,0 +1,179 @@
+        rl.release()
+        rl.release()
+
+
+class TestThreadSafety:
+    """Basic smoke test for concurrent usage."""
+
+    def test_concurrent_acquires(self):
+        rl = RateLimiter(max_per_second=50, max_per_minute=500, max_concurrent=5)
+        errors: list[Exception] = []
+
+        def worker():
```

Ruff (0.15.9)

[warning] Missing return type annotation for private function `worker`

Add return type annotation: `None`

(ANN202)